test-data/unit/check-incremental.test

```diff
@@ -908,3 +1193,309 @@ foo(3)
+# cmd2: mypy -m main
+# flags: --silent-imports
+
+[file main.py]
+from foo import bar
+def accept_int(x: int) -> None: pass
+accept_int(bar)
+
+[file foo.py]
+bar = 3
+
+[file foo.py.next]
+# Empty!
+
+[rechecked main]
+[stale main]
+
+[case testIncrementalSilencedModuleNoLongerCausesError]
+# cmd: mypy -m main evil
+# cmd2: mypy -m main
+# flags: --silent-imports
+
+[file main.py]
+from evil import bar
+def accept_int(x: int) -> None: pass
+accept_int(bar)
+reveal_type(bar)
+
+[file evil.py]
+bar = "str"
+
+[rechecked main]
+[stale]
+[out1]
+tmp/main.py:3: error: Argument 1 to "accept_int" has incompatible type "str"; expected "int"
+tmp/main.py:4: error: Revealed type is 'builtins.str'
+[out2]
+tmp/main.py:4: error: Revealed type is 'Any'
+
+[case testIncrementalFixedBugCausesPropagation]
+import mod1
+
+[file mod1.py]
+from mod2 import A
+val = A().makeB().makeC().foo()
+reveal_type(val)
+
+[file mod2.py]
+from mod3 import B
+class A:
+    def makeB(self) -> B: return B()
+
+[file mod3.py]
+from mod4 import C
+class B:
+    def makeC(self) -> C:
+        val = 3  # type: int
+        val = "str"   # deliberately triggering error
+        return C()
+
+[file mod3.py.next]
+from mod4 import C
+class B:
+    def makeC(self) -> C: return C()
+
+[file mod4.py]
+class C:
+    def foo(self) -> int: return 1
+
+[rechecked mod3, mod2, mod1]
+[stale mod3, mod2]
+[out1]
+tmp/mod2.py:1: note: In module imported here,
+tmp/mod1.py:1: note: ... from here,
+main:1: note: ... from here:
+tmp/mod3.py: note: In member "makeC" of class "B":
+tmp/mod3.py:5: error: Incompatible types in assignment (expression has type "str", variable has type "int")
+main:1: note: In module imported here:
+tmp/mod1.py: note: At top level:
+tmp/mod1.py:3: error: Revealed type is 'builtins.int'
+
+[out2]
+main:1: note: In module imported here:
+tmp/mod1.py:3: error: Revealed type is 'builtins.int'
+
+[case testIncrementalIncidentalChangeWithBugCausesPropagation]
+import mod1
+
+[file mod1.py]
+from mod2 import A
+val = A().makeB().makeC().foo()
+reveal_type(val)
+
+[file mod2.py]
+from mod3 import B
+class A:
+    def makeB(self) -> B: return B()
+
+[file mod3.py]
+from mod4 import C
+class B:
+    def makeC(self) -> C:
+        val = 3  # type: int
+        val = "str"   # deliberately triggering error
+        return C()
+
+[file mod4.py]
+class C:
+    def foo(self) -> int: return 1
+
+[file mod4.py.next]
+class C:
+    def foo(self) -> str: return 'a'
+
+[rechecked mod4, mod3, mod2, mod1]
+[stale mod4]
+[out1]
+tmp/mod2.py:1: note: In module imported here,
+tmp/mod1.py:1: note: ... from here,
+main:1: note: ... from here:
+tmp/mod3.py: note: In member "makeC" of class "B":
+tmp/mod3.py:5: error: Incompatible types in assignment (expression has type "str", variable has type "int")
+main:1: note: In module imported here:
+tmp/mod1.py: note: At top level:
+tmp/mod1.py:3: error: Revealed type is 'builtins.int'
+
+[out2]
+tmp/mod2.py:1: note: In module imported here,
```

Comment:
I tried using the flags, and unfortunately found that they didn't do a very good job of suppressing the notes. In this case, for example, the output for one run ended up shrinking to only:

```
  tmp/mod2.py:1: note: In module imported here,
  tmp/mod1.py:1: note: ... from here,
  main:2: note: ... from here:                  (diff)
  tmp/mod3.py:5: error: Incompatible types in assignment (expression has type "str", variable has type "int") (diff)
  main:2: note: In module imported here:        (diff)
  tmp/mod1.py:3: error: Revealed type is 'builtins.int' (diff)
```

...which is only two lines shorter, so it didn't really seem to me that adding the flag was worth it.